packages/core/src/components/select/select.tsx

```diff
@@ -349,32 +349,49 @@ export class Select {
     }
   }
 
-  private onArrowNavigation(event: KeyboardEvent) {
-    event.stopPropagation();
+  private onArrowNavigation(
+    event: KeyboardEvent,
+    key: 'ArrowDown' | 'ArrowUp'
+  ) {
     event.preventDefault();
+    event.stopPropagation();
 
-    const focusItem = this.items.find(
-      (item) => document.activeElement === item.querySelector('button')
-    );
-    this.navigationItem = focusItem;
+    this.dropdownShow = true;
 
-    const selectItems = this.items.filter(
-      (i) => !i.classList.contains('d-none')
+    const items = this.items.filter((i) => !i.classList.contains('d-none'));
+    if (this.navigationItem === undefined) {
+      this.applyFocusTo(items[0]);
+      return;
+    }
+
+    const indexOfNavigationItem = items.findIndex(
```

Comment:
```suggestion
    let newIndex = items.findIndex(
```

And delete line 371?